src/Tizen.Diagnostics/Tizen.Diagnostics/Diagnostics.cs

```diff
@@ -0,0 +1,304 @@
+        ~DiagnosticsContext()
+        {
+            Dispose(false);
+        }
+
+        /// <summary>
+        /// Gets report of specific type from the context.
+        /// </summary>
+        /// <since_tizen> 8 </since_tizen>
+        /// <param name="report">Report type to retrieve from the context.</param>
+        /// <returns>
+        /// The instance of DiagnosticsData.
+        /// </returns>
+        /// <privilege>
+        /// http://tizen.org/privilege/internal/default/platform
+        /// </privilege>
+        public DiagnosticsData GetReport(ReportType report)
+        {
+            var ret = NativeApi.GetReport(_handle, report, out IntPtr data);
+            if (ret != NativeApi.DiagnosticsError.None)
+                throw ExceptionFactory.CreateException(ret);
+
+            return new DiagnosticsData(data);
+        }
+
+        /// <summary>
+        /// Stores context type.
+        /// </summary>
+        /// <since_tizen> 8 </since_tizen>
+        public ContextType Type { get; internal set; }
+
+        /// <summary>
+        /// Dispose API for closing the internal resources.
+        /// </summary>
+        /// <since_tizen> 8 </since_tizen>
+        public void Dispose()
+        {
+            Dispose(true);
+            GC.SuppressFinalize(this);
+        }
+
+        /// <summary>
+        /// Dispose API for closing the internal resources.
+        /// </summary>
+        /// <since_tizen> 8 </since_tizen>
+        protected virtual void Dispose(bool disposing)
+        {
+            if (!_disposed)
+            {
+                if (_handle != IntPtr.Zero)
+                {
+                    NativeApi.Destroy(_handle);
+                    _handle = IntPtr.Zero;
+                }
+                _disposed = true;
+            }
+        }
+    }
+
+    /// <summary>
+    /// The DiagnosticsData class is used for reading data (either from report or client).
+    /// </summary>
+    /// <since_tizen> 8 </since_tizen>
+    public class DiagnosticsData : IDisposable
+    {
+        private IntPtr _handle = IntPtr.Zero;
+        private bool _disposed = false;
+
+        /// <summary>
+        /// The DiagnosticsData class constructor.
+        /// </summary>
+        /// <since_tizen> 8 </since_tizen>
+        /// <param name="handle">Diagnostics data handle.</param>
+        internal DiagnosticsData(IntPtr handle)
+        {
+            _handle = handle;
+        }
+
+        /// <summary>
+        /// Finalizes an instance of the DiagnosticsData class.
+        /// </summary>
+        ~DiagnosticsData()
+        {
+            Dispose(false);
+        }
+
+        /// <summary>
+        /// Reads given number of bytes from diagnostics data.
+        /// </summary>
+        /// <since_tizen> 8 </since_tizen>
+        /// <param name="size">Number of bytes to read.</param>
+        /// <returns>
+        /// An array of bytes read.
+        /// </returns>
+        public byte[] Read(int size)
```

Comment:
How do users know the size to read?

Reply:
This function should be used in loop reading pieces of data until EOF (returns null). The size may (crash report) or may not (client's data) be known, so there is no API to check it.

Reply:
In that case, It is not good solution to return a byte array allocated in the method. It will make many allocation and copy in the real use-case. 
Please change the signature to
```c#
int Read(Byte[] buffer, int count)
```
if `offset` is possible, 
```c#
int Read(Byte[] buffer, int offset, int count)
```
See: https://docs.microsoft.com/ko-kr/dotnet/api/system.io.stream.read?view=netcore-3.1
And please add more user-kindly description to these methods.

Reply:
Done. Unfortunately, offset is not possible.

Reply:
offsetInt32
The zero-based byte offset in buffer at which to begin storing the data read from the current stream.

Why not support? it is not related with native api

Reply:
Ok, I thought about offset in reading from source, not in writing to buffer.

Reply:
@myroot It follows that below functions are needed for reading:
        public int Read(byte[] buffer, int count);
        public int Read(byte[] buffer, int count, int timeout);
        public int Read(byte[] buffer, int offset, int count);
        public int Read(byte[] buffer, int offset, int count, int timeout);
There is a conflict in signatures. How you prefer to resolve it?

Reply:
``` c#
int Read(byte[] buffer, int offset, int count);
int Read(byte[] buffer, int offset, int count, int timeout);
```

Reply:
ok, sure.

Reply:
Done.